docs/src/stake-pool.md

```diff
@@ -1,11 +1,427 @@
 ---
-title: Stake-Pool Program
+title: Stake Pool Program
 ---
 
-A program for pooling together SOL to be staked by an off-chain agent running SoM (Stake-o-Matic).
+A program for pooling together SOL to be staked by an off-chain agent running
+a Delegation bot which redistributes the stakes across the network and tries
+to maximize censorship resistance and rewards.
 
-Each SoM needs at least one pool.  Users deposit stakes into the SoM pool
-and receives a pool token minus the fee.  The SoM redistributes the stakes
-across the network and tries to maximize censorship resistance and rewards.
+## Overview
 
-The project is under construction.
\ No newline at end of file
+SOL token holders can earn rewards and help secure the network by staking tokens
+to one or more validators. Rewards for staked tokens are based on the current
+inflation rate, total number of SOL staked on the network, and an individual 
+validator’s uptime and commission (fee).
+
+Stake pools are an alternative method of earning staking rewards. This on-chain
+program pools together SOL to be staked by a manager, allowing SOL holders to
+stake and earn rewards without managing stakes.
+
+Additional information regarding staking and stake programming is available at:
+
+- https://solana.com/staking
+- https://docs.solana.com/staking/stake-programming
+
+
+## Background
+
+Solana's programming model and the definitions of the Solana terms used in this
+document are available at:
+
+- https://docs.solana.com/apps
+- https://docs.solana.com/terminology
+
+## Source
+
+The Stake Pool Program's source is available on
+[github](https://github.com/solana-labs/solana-program-library).
+
+## Operational overview
+
+The following explains the instructions available in the Stake Pool Program along
+with examples using the command-line utility.
+
+## Very important note concerning activated stakes
+
+In its current iteration, the stake pool only processes totally active stakes.
+Deposits must come from fully active stakes, and withdrawals return a fully 
+active stake account.
+
+Also, the deposited stake account's "credits observed" must match the destination
+account's "credits observed". Typically, this means you must wait an additional
+epoch after activation for your stake account to match up with the stake pool's account.
+
+This feature maintains fungibility of stake pool tokens. Fully activated stakes
+are not equivalent to inactive, activating, or deactivating stakes due to the
+time cost of staking. Otherwise, malicious actors can deposit stake in one state
+and withdraw it in another state without waiting.
+
+Future iterations of the stake pool program aim to improve on this limitation.
+
+## Command-line Utility
+
+The `spl-stake-pool` command-line utility can be used to experiment with SPL
+tokens.  Once you have [Rust installed](https://rustup.rs/), run:
+```sh
+$ cargo install spl-stake-pool-cli
```

Comment:
Missing pre-requisites for this cargo command. In my case, I see this:
```
% cargo install spl-stake-pool-cli
    Updating crates.io index
error: could not find `spl-stake-pool-cli` in registry `https://github.com/rust-lang/crates.io-index` with version `*`
```

Reply:
Sorry about that, the crate hasn't been published yet, so you'll have to build / run things yourself.  Is that ok?

Reply:
No worries for me, I'm sure I can figure it out. A non-technical user will see this as a road-block. A quick mention of pre--reqs near the top might help the reader know what to expect.